cfme/utils/appliance/__init__.py

```diff
@@ -2174,71 +2176,108 @@ def check_domain_enabled(self, domain):
+        assert self.ssh_client.run_command('echo "{s}" > {c}'
+                                           .format(s=auth_provider.sssd_conf,
+                                                   c=self.CONF_FILES['sssd']))
         assert self.ssh_client.run_command('chown -R root:root {}').format(self.CONF_FILES['sssd'])
         assert self.ssh_client.run_command('chmod 600 {}').format(self.CONF_FILES['sssd'])
         # copy miq/cfme template files for httpd ext auth config
-        template_dir = self.CONF_FILES.get(
-            'downstream_templates' if self.is_downstream else 'upstream_templates'
-        )
+        template_dir = self.CONF_FILES.get('downstream_templates'
+                                           if self.is_downstream
+                                           else 'upstream_templates')
         # pam httpd-auth and httpd remote-user.conf
-        for file in [self.CONF_FILES['pam_httpd_auth'], self.CONF_FILES['httpd_remote_user']]:
-            assert self.ssh_client.run_command('cp {t}{c} {c}'.format(t=template_dir, c=file))
+        for conf_file in [self.CONF_FILES['pam_httpd_auth'], self.CONF_FILES['httpd_remote_user']]:
+            assert self.ssh_client.run_command('cp {t}{c} {c}'.format(t=template_dir, c=conf_file))
 
         # https external-auth conf, template has extra '.erb' suffix
-        assert self.ssh_client.run_command(
-            'cp {t}{c}.erb {c}'
-            .format(t=template_dir, c=self.CONF_FILES['httpd_ext_auth'])
-        )
+        assert self.ssh_client.run_command('cp {t}{c}.erb {c}'
+                                           .format(t=template_dir,
+                                                   c=self.CONF_FILES['httpd_ext_auth']))
         assert self.ssh_client.run_command('setenforce 0')
         self.sssd.restart()
         self.httpd.restart()
         self.wait_for_web_ui()
 
         # UI configuration of auth provider type
-        self.server.authentication.configure_auth(auth_mode='external', **kwargs)
+        self.server.authentication.configure(auth_mode='external', auth_provider=auth_provider)
 
     @logger_wrap('Disabling openldap external auth provider')
-    def disable_openldap(self):
+    def disable_openldap(self, log_callback=None):
         self.server.authentication.configure_auth()
         files_to_remove = [
-            '/etc/sssd/sssd.conf',
-            '/etc/pam.d/httpd-auth',
-            self.HTTPD_CONF_MODULES['manageiq_ext_auth'],
-            self.HTTPD_CONF_MODULES['manageiq_remoteuser']
+            self.CONF_FILES['sssd'],
+            self.CONF_FILES['pam_httpd_auth'],
+            self.CONF_FILES['httpd_ext_auth'],
+            self.CONF_FILES['httpd_remote_user']
         ]
-        for file in files_to_remove:
-            assert self.ssh_client.run_command('rm -rf {}'.format(file))
+        for conf_file in files_to_remove:
+            assert self.ssh_client.run_command('rm -rf {}'.format(conf_file))
```

Comment:
That -r is scary.   moving along.

Reply:
![image](https://user-images.githubusercontent.com/21315076/37305506-0d47e0a0-260b-11e8-9c02-b48255be4983.png)

Reply:
![nothing-to-see-here-gif-4](https://user-images.githubusercontent.com/21315076/37305538-24d8eb24-260b-11e8-81c8-43f2e1c0c963.gif)

Reply:
Which directory are we in for this?

Reply:
`files_to_remove` is a list of `self.CONF_FILES` values, which _should_ all be complete file paths, but yeah this is risky and probably doesn't need the recurse flag, just the force flag.

Reply:
Also wrapping an `ls` into the rm command so it only calls against files that exist.